### PR TITLE
IP address and policy removal on connection update

### PR DESCRIPTION
**Pull request by glazychev-art**
Сontinuation of: https://github.com/networkservicemesh/sdk-kernel/pull/433

Signed-off-by: Artem Glazychev <artem.glazychev@xored.com>

### CODE REVIEW

**Comment by glazychev-art** [comment]
@LionelJouin @denis-tingaikin @edwarnicke @NikitaSkrynnik 

This is continuation of: https://github.com/networkservicemesh/sdk-kernel/pull/433
We added a dump for iptables here because of: https://github.com/networkservicemesh/sdk-kernel/pull/433#issuecomment-1077523667

@LionelJouin Could you please check, if this solves your problems?

**Review by denis-tingaikin — APPROVED** [review]

File: pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go [hunk 1]
```diff
@@ -232,28 +239,40 @@ func routeAdd(ctx context.Context, handle *netlink.Handle, l netlink.Link, route
...
+		ps, ok := tableIDs.Load(conn.GetId())
+		if ok {
+			for tableID, policy := range ps {
+				if err := delRule(ctx, netlinkHandle, policy, tableID); err != nil {
+					return errors.WithStack(err)
+				}
+				delete(ps, tableID)
+				tableIDs.Store(conn.GetId(), ps)
+			}
+			if len(ps) == 0 {
```
Comment (LionelJouin) [inline pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go:258]: I think this is no longer needed

**Review by LionelJouin — COMMENTED** [review]
I just tried it, it works fine

File: pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go (hunk 1 above)
Comment (glazychev-art) [reply pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go]: Yes, thank you. I simplified a little

**Review by LionelJouin — APPROVED** [review]

**Pull request merged**